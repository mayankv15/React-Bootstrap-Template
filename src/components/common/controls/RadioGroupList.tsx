import React from "react";
import {
  FormLabel,
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup as MUIRadioGroup,
} from "@mui/material";
import { makeStyles } from "@mui/styles";

import SIZE from "../../../theme/font_size.json";
import COLORS from "../../../theme/colors.json";
import { CheckCircle, RadioButtonUnchecked } from "@mui/icons-material";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles((theme) => ({
  groupRadioWrapper: {
    "& .MuiFormControl-root": {
      width: "100%",
    },
  },
  radioGroupList: {
    "& .MuiRadio-root": {
      padding: "0 9px",
    },
    "& .MuiFormControlLabel-root": {
      color: theme.palette.primary.main,
    },
    "& .MuiSvgIcon-root": {
      backgroundColor: COLORS.WHITE,
    },
    "& .MuiRadio-colorSecondary.Mui-checked": {
      color: theme.palette.primary.main,
    },
  },
  custRadioGroupList: {
    "& .MuiFormLabel-root": {
      color: COLORS.GRAY,
      fontSize: SIZE[12],
      marginBottom: SIZE[8],
    },
    "& .MuiFormGroup-row": {
      "& .MuiFormControlLabel-root": {
        marginRight: SIZE[50],
      },
    },
    "& .MuiFormControlLabel-root": {
      "& .MuiTypography-root": {
        fontSize: SIZE[14],
      },
      "& .Mui-checked + .MuiTypography-root": {
        color: theme.palette.primary.main,
      },
    },
  },
}));

interface RadioItemProps {
  enabled?: boolean;
  label: string;
  key: string;
  col?: number;
  description?: string;
  isActive?: boolean;
}

interface LabelProps {
  label?: any;
  name?: string;
  list: RadioItemProps[];
  value?: string;
  required?: boolean;
  row?: boolean;
  className?: string;
  handleChange?: (event) => void;
}
const RadioGroupList = ({
  list,
  name,
  label,
  value,
  className,
  handleChange,
  required,
  row = true,
}: LabelProps) => {
  const classes = useStyles();
  const { t } = useTranslation();
  const renderRadioList = (item) => {
    return (
      <FormControlLabel
        label={t(item.label)}
        value={item.key}
        disabled={!item.enabled}
        control={
          <Radio
            checked={value === item.key}
            icon={<RadioButtonUnchecked />}
            checkedIcon={<CheckCircle />}
          />
        }
      />
    );
  };

  return (
    <div className={classes.custRadioGroupList + " custRadioGroupList"}>
      <FormControl
        component="fieldset"
        style={{ width: "100%" }}
        className={className}
      >
        <FormLabel component="legend">
          {t(label)} {required ? "*" : null}
        </FormLabel>
        <MUIRadioGroup
          row={row}
          aria-label={label}
          name={name}
          className={classes.radioGroupList}
          value={value}
          onChange={handleChange}
        >
          {list &&
            Array.isArray(list) &&
            list.map((li, index) => (
              <div key={index}>{renderRadioList(li)}</div>
            ))}
        </MUIRadioGroup>
      </FormControl>
    </div>
  );
};

export default RadioGroupList;
