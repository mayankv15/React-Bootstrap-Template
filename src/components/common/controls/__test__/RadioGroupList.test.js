import React from "react";
import { render } from "@testing-library/react";
import RadioGroupList from "../RadioGroupList"

describe("Renders RadioGroupList", () => {
    it("should render RadioGroupList correctly", () => {
        const GENDER = [
            {
                key: "MALE",
                label: "Male",
                enabled: true,
                parentKey: null,
            },
            {
                key: "FEMALE",
                label: "Female",
                enabled: true,
                parentKey: null,
            },
        ];

        const component = render(<RadioGroupList
            required={true}
            label={"GENDER"}
            name="gender"
            list={GENDER}
            row={false} />);
        expect(component).toMatchSnapshot();
    });
});
