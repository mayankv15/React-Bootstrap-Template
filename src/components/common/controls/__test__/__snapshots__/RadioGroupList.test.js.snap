// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RadioGroupList should render RadioGroupList correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-custRadioGroupList-3 custRadioGroupList"
      >
        <fieldset
          class="MuiFormControl-root"
          style="width: 100%;"
        >
          <legend
            class="MuiFormLabel-root"
          >
            GENDER
             
            *
          </legend>
          <div
            aria-label="GENDER"
            class="MuiFormGroup-root makeStyles-radioGroupList-2"
            role="radiogroup"
          >
            <div>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-7"
                      name="gender"
                      type="radio"
                      value="MALE"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Male
                </span>
              </label>
            </div>
            <div>
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-7"
                      name="gender"
                      type="radio"
                      value="FEMALE"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Female
                </span>
              </label>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-custRadioGroupList-3 custRadioGroupList"
    >
      <fieldset
        class="MuiFormControl-root"
        style="width: 100%;"
      >
        <legend
          class="MuiFormLabel-root"
        >
          GENDER
           
          *
        </legend>
        <div
          aria-label="GENDER"
          class="MuiFormGroup-root makeStyles-radioGroupList-2"
          role="radiogroup"
        >
          <div>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-7"
                    name="gender"
                    type="radio"
                    value="MALE"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Male
              </span>
            </label>
          </div>
          <div>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-7"
                    name="gender"
                    type="radio"
                    value="FEMALE"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Female
              </span>
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
